charts/spire/charts/spire-server/templates/hpa.yaml

```diff
@@ -9,24 +9,28 @@ metadata:
 spec:
   scaleTargetRef:
     apiVersion: apps/v1
-    kind: Deployment
+    kind: StatefulSet
```

Comment:
Ouch that is a bug for a while. Good catch.

Reply:
hmm.....

If this is the case, the chart on k8s 1.21 / 1.22 has never worked with hpa.... they introduced apps/v2 in k8s 1.23.
Maybe we should just support v2 on this object since its very unlikely to affect anyone at 1.21 (Won't affect my 1.21 cluster)

Reply:
Didn't we switch spire-server a while ago to a Statefulset. Originally it was a Deployment.

Reply:
It could have easily been a Deployment in the very beginning, yeah.